model/DataList.php

```diff
@@ -1075,70 +1082,19 @@ public function removeByID($itemID) {
 	 * @return DataList
 	 */
 	public function reverse() {
-		return $this->alterDataQuery(function($query){
+		return $this->alterDataQuery(function(DataQuery $query){
 			$query->reverseSort();
 		});
 	}
 
 	/**
-	 * This method won't function on DataLists due to the specific query that it represent
-	 *
-	 * @param mixed $item
-	 */
-	public function push($item) {
```

Comment:
As a note, I've also removed the overrides in ManyManyList that refers to DataList methods that it didn't support, so this is just the other half of that cleanup.